apps/ios/SimpleX NSE/NotificationService.swift

```diff
@@ -137,24 +223,48 @@ class NSEThreads {
         }
     }
 
+    @inline(__always)
     var noThreads: Bool {
         allThreads.isEmpty
     }
 }
 
-struct ExpectedMessage {
+// ExpectedMessage is a bad name, because it containes expectedMsg inside.
+// It is the state for a processor for connection entity that expects a message.
+// Single NSE instance may have several of them.
+struct NotificationEntity {
     var ntfConn: NtfConn
-    var expectedMsgId: String?
-    var allowedGetNextAttempts: Int
+    var entityId: ChatId
+    var expectedMsg: NtfMsgInfo?
+    var allowedGetNextAttempts: Int = 3
     var msgBestAttemptNtf: NSENotificationData?
-    var ready: Bool
-    var shouldProcessNtf: Bool
-    var startedProcessingNewMsgs: Bool
-    var semaphore: DispatchSemaphore
+
+    // startedProcessingNewMsgs determines that the entity stared processing events once it processed dropped notifications.
+    // It remains true when shouldProcessNtf is set to false, to prevent NSE from being chosen as the next for the entity.
+    // It is atomically set to true by startThead or by nextThread
+    var startedProcessingNewMsgs: Bool = false
+
+    // shouldProcessNtf determines that NSE should process events for this entity,
+    // it is atomically set:
+    // - to true in processDroppedNotifications in case dropped notification is not delivered, and more messages needed
```

Comment:
```suggestion
    // - to true in processDroppedNotifications in case dropped notification is not chosen for delivery, and more messages are needed.
```